File: src/bindings/java/in3/eth1/LogFilter.java
package in3.eth1;

import in3.JSON;

/**
 * Log configuration for search logs.
 */

public class LogFilter {

    /**
     * Quantity or Tag - (optional) (default: latest) Integer block number, or
     * 'latest' for the last mined block or 'pending', 'earliest' for not yet mined
     * transactions.
     */
    long fromBlock = Block.LATEST;
    /**
     * Quantity or Tag - (optional) (default: latest) Integer block number, or
     * 'latest' for the last mined block or 'pending', 'earliest' for not yet mined
     * transactions.
     */
    long toBlock = Block.LATEST;
    /**
     * (optional) 20 Bytes - Contract address or a list of addresses from which logs
     * should originate.
     */
    String address;
    /**
     * (optional) Array of 32 Bytes Data topics. Topics are order-dependent. It’s
     * possible to pass in null to match any topic, or a subarray of multiple topics
     * of which one should be matching.
     */
    Object[] topics;

    /** a(optional) The maximum number of entries to retrieve (latest first). */
    int limit;

    /**
     * creates a JSON-String.
     */
    public String toString() {
        StringBuilder sb = new StringBuilder('{');
        if (fromBlock >= 0)
            JSON.appendKey(sb, "fromBlock", fromBlock);
        if (toBlock >= 0)
            JSON.appendKey(sb, "toBlock", toBlock);
        if (topics != null)
            JSON.appendKey(sb, "topics", JSON.toJson(topics));
        if (limit > 0)
            JSON.appendKey(sb, "limit", JSON.asString(limit));
        if (address != null)
            JSON.appendKey(sb, "address", JSON.asString(address));
        sb.setCharAt(sb.length() - 1, '}');
        return sb.toString();
    }

}

File: dotnet/In3/Eth1/LogFilter.cs
using System.Numerics;
namespace In3.Eth1
{
    /// <summary>
    /// Filter configuration for search logs. To be used along with the <see cref="Eth1.Api" /> filter and methods.
    /// </summary>
    public class LogFilter
    {

        /// <summary>
        /// Starting block for the filter.
        /// </summary>
        public BigInteger? FromBlock { get; set; }

        /// <summary>
        /// End block for the filter.
        /// </summary>
        public BigInteger? ToBlock { get; set; }

        /// <summary>
        /// Address for the filter.
        /// </summary>
        public string Address { get; set; }

        /// <summary>
        /// Array of 32 Bytes Data topics. Topics are order-dependent. It's  possible to pass in null to match any topic, or a subarray of multiple topics  of which one should be matching.
        /// </summary>
        public object[] Topics { get; set; }

        /// <summary>
        /// Blcok hash of the filtered blocks.
        /// </summary>
        /// <remarks>If present, <see cref="FromBlock" /> and <see cref="ToBlock" /> will be ignored.</remarks>
        public string BlockHash { get; set; }

        /// <summary>
        /// Standard constructor.
        /// </summary>
        public LogFilter() { }

        internal Rpc.LogFilter ToRPc()
        {
            Rpc.LogFilter result = new Rpc.LogFilter();
            if (FromBlock != null) result.FromBlock = BlockParameter.AsString(FromBlock.Value);
            if (ToBlock != null) result.ToBlock = BlockParameter.AsString(ToBlock.Value);
            result.Address = Address;
            result.Topics = Topics;
            result.BlockHash = BlockHash;

            return result;
        }
    }
}